guides/micronaut-cloud-secrets-azure/micronaut-cloud-secrets-azure.adoc

```diff
@@ -0,0 +1,194 @@
+
+Add the following dependency:
+
+:dependencies:
+
+dependency:micronaut-azure-secret-manager[groupId=io.micronaut.azure]
+
+:dependencies:
+
+== Update the Application configuration to use Azure Key Vault secrets
+
+=== Add the Azure Key Vault secret placeholders to the Application Configuration
+[,diff]
+.src/main/resources/application.yml
+----
+micronaut:
+  application:
+    name: micronautguide
+
+datasources:
+  default:
++   username: ${JDBC-USER}
++   password: ${JDBC-PASSWORD}
+    dialect: MYSQL
+    driverClassName: ${JDBC_DRIVER:com.mysql.cj.jdbc.Driver}
+
+flyway:
+  datasources:
+    default:
+      enabled: true
+----
+
+NOTE: The Azure Key Vault secrets naming convention does not allow "_" to be used as a word joiner. Therefore, when adding JDBC secret placeholders, you should use "-" instead (e.q. JDBC-USER, JDBC-PASSWORD).
+
+=== Enable Distributed Configuration
+
+This is accomplished by creating a `bootstrap.yml` file in the `resources` directory and adding the following:
+
+[,yml]
+.src/main/resources/bootstrap.yml
+----
+micronaut:
+  application:
+    name: micronautguide # <1>
+  config-client:
+    enabled: true # <2>
+
+azure:
+  credential:
+    client-secret:
+      client-id: ${AZURE_CLIENT_ID} # <3>
+      tenant-id: ${AZURE_TENANT_ID} # <4>
+      secret: ${AZURE_SECRET_VALUE} # <5>
+  key-vault:
+    vault-url: ${AZURE_VAULT_URI} # <6>
+----
+<1> The application name entry is moved from `src/main/resources/application.yml` for resolving distributed configurations properties
+<2> Flag to enable the distributed configurations
+<3> The client ID of the trusted application that you will register in Azure Portal
+<4> The tenant ID
+<5> The secret value of the registered application
+<6> The Azure Key Vault resource URI
+
+NOTE: More details on how to obtain these values are provided in the Azure Portal section below.
+
+https://docs.micronaut.io/latest/guide/#bootstrap[Click here to get more details on distributed configuration]
+
+=== Clean up of the Application Configuration
+
+Remove the redundant application name entry from the `application.yml`.
+
+[,diff]
+.src/main/resources/application.yml
+----
+- micronaut:
+-  application:
+-    name: micronautguide
+
+datasources:
+  default:
+    username: ${JDBC-USER}
+    password: ${JDBC-PASSWORD}
+    dialect: MYSQL
+    driverClassName: ${JDBC_DRIVER:com.mysql.cj.jdbc.Driver}
+
+flyway:
+  datasources:
+    default:
+      enabled: true
+----
+
+== Store JDBC secrets in Azure Key Vault using Azure Portal
+
+Create Azure Key Vault service resource.
+{empty} +
+
+image::azure-key-vault/azure-vault-create-vault-resouce.png[]
+{empty} +
+
+When Key Vault service deployment is ready, please go to the resource in order to create secrets. Prior to clicking on the Secrets option in the left menu list, please copy the Vault URI from the main screen section. You will need to set it as an environment variable in order to integrate Azure Key Vault into your application.
```

Comment:
you can remove the use of the word "please" in this paragraph

Reply:
Will do.

Reply:
Covered by https://github.com/micronaut-projects/micronaut-guides/pull/1099/commits/8fed9a79fe5adb0859856581038fe14bef237f02.